common/oatbox/extension/Manifest.php

```diff
@@ -0,0 +1,411 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\oatbox\extension;
+
+use oat\oatbox\extension\exception\ManifestException;
+use oat\oatbox\extension\exception\ManifestNotFoundException;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+
+/**
+ * Class Manifest
+ * @author Jerome Bogaerts <jerome@taotesting.com>
+ * @package oat\oatbox\extension
+ */
+class Manifest implements ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    /**
+     * The path to the file where the manifest is described.
+     * @var string
+     */
+    private $filePath = '';
+
+    /**
+     * The version of the Extension the manifest describes.
+     * @var string
+     */
+    private $version = null;
+    
+    /**
+     * The dependencies of the Extension the manifest describes.
+     * @var array
+     */
+    private $dependencies = [];
+
+    /**
+     * @var array
+     */
+    private $manifest = [];
+
+    /**
+     * @var ComposerInfo
+     */
+    private $composerInfo = null;
+
+    /**
+     * Creates a new instance of Manifest.
+     *
+     * @access public
+     * @param string $filePath The path to the manifest.php file to parse.
+     * @param ComposerInfo $composerInfo
+     * @throws ManifestNotFoundException
+     * @throws exception\MalformedManifestException
+     */
+    public function __construct($filePath, ComposerInfo $composerInfo = null)
```

Comment:
I think it would be better to pass directory name instead of manifest file path here. This class is more suitable for manifest fine name constant, than `common_ext_Extension` and later in the code we use directory name to read composer files.